testdata/p4_16_samples/checksum-l4-bmv2.p4

```diff
@@ -0,0 +1,360 @@
+/*
+ * Copyright 2020 MNK Labs & Consulting, LLC.
+ *
+ * https://mnkcg.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Jointly developed with Andy Fingerhut (ex-Cisco and Barefoot/Intel).
+ *
+ * This program does not support IPv6.  However, IPv6 also uses
+ * a pseudo header for upper-layer protocols (e.g., TCP and UDP).
+ * More details for IPv6 checksum is available at the link below.
+ *
+ * https://tools.ietf.org/html/rfc8200#page-27
+ *
+ * The program does not support UDP options because none have been specified
+ * in a RFC.
+*/
+
+#include <core.p4>
+#include <v1model.p4>
+
+typedef bit<48>  EthernetAddress;
+typedef bit<32>  IPv4Address;
+
+header ethernet_t {
+    bit<48> dstAddr;
+    bit<48> srcAddr;
+    bit<16> etherType;
+}
+
+// IPv4 header _with_ options
+header ipv4_t {
+    bit<4>       version;
+    bit<4>       ihl;
+    bit<8>       diffserv;
+    bit<16>      totalLen;
+    bit<16>      identification;
+    bit<3>       flags;
+    bit<13>      fragOffset;
+    bit<8>       ttl;
+    bit<8>       protocol;
+    bit<16>      hdrChecksum;
+    IPv4Address  srcAddr;
+    IPv4Address  dstAddr;
+    varbit<320>  options;
+}
+
+header tcp_t {
+    bit<16> srcPort;
+    bit<16> dstPort;
+    bit<32> seqNo;
+    bit<32> ackNo;
+    bit<4>  dataOffset;
+    bit<3>  res;
+    bit<3>  ecn;
+    bit<6>  ctrl;
+    bit<16> window;
+    bit<16> checksum; // Includes Pseudo Hdr + TCP segment (hdr + payload)
+    bit<16> urgentPtr;
+    varbit<320>  options;
+}
+
+header udp_t {
+    bit<16> srcPort;
+    bit<16> dstPort;
+    bit<16> length_;
+    bit<16> checksum;
+}
+
+header IPv4_up_to_ihl_only_h {
+    bit<4>       version;
+    bit<4>       ihl;
+}
+
+header tcp_upto_data_offset_only_h {
+    bit<16> srcPort;
+    bit<16> dstPort;
+    bit<32> seqNo;
+    bit<32> ackNo;
+    // dataOffset in TCP hdr uses 4 bits but set to 8 bits below.
+    // If 4 bits are used for it, p4c-bm2-ss complains the header
+    // is not a multiple of 8 bits. In the lookahead we use bit Slice
+    // to get 4 bits.
+    bit<8>  dataOffset;
+}
```

Comment:
You could also use something like this, which would avoid need for slice after lookahead:
```
    bit<4> dataOffset;
    bit<4> dontCare;
```
since you really do not care what the contents of any of the fields are except `dataOffset` for this header that is only used for lookahead purposes.